app/controllers/declaration/AdditionalFiscalReferencesController.scala

```diff
@@ -40,83 +41,104 @@ class AdditionalFiscalReferencesController @Inject()(
+          updateCacheModels(itemId, updatedCache, routes.AdditionalFiscalReferencesController.displayPage(itemId))
       )
 
-  private def saveAndContinue(form: Form[AdditionalFiscalReference], cachedData: AdditionalFiscalReferencesData)(
-    implicit request: JourneyRequest[_]
-  ): Future[Result] =
+  private def saveAndContinue(
+    itemId: String,
+    form: Form[AdditionalFiscalReference],
+    cachedData: AdditionalFiscalReferencesData
+  )(implicit request: JourneyRequest[_]): Future[Result] =
     MultipleItemsHelper
-      .saveAndContinue(form, cachedData.references, true, limit)
+      .saveAndContinue(form, cachedData.references, isMandatory = true, limit)
       .fold(
-        formWithErrors => Future.successful(badRequest(formWithErrors, cachedData.references)),
+        formWithErrors => Future.successful(badRequest(itemId, formWithErrors, cachedData.references)),
         updatedCache =>
           if (updatedCache != cachedData.references)
-            customsCacheService
-              .cache[AdditionalFiscalReferencesData](
-                goodsItemCacheId(),
-                formId,
-                AdditionalFiscalReferencesData(updatedCache)
-              )
-              .map(_ => Redirect(routes.ItemTypePageController.displayPage()))
-          else Future.successful(Redirect(routes.ItemTypePageController.displayPage()))
+            updateCacheModels(itemId, updatedCache, routes.ItemTypePageController.displayPage(itemId))
+          else Future.successful(Redirect(routes.ItemTypePageController.displayPage(itemId)))
       )
 
-  private def removeReference(values: Seq[String], cachedData: AdditionalFiscalReferencesData)(
+  private def removeReference(itemId: String, values: Seq[String], cachedData: AdditionalFiscalReferencesData)(
     implicit request: JourneyRequest[_]
   ): Future[Result] = {
     val updatedCache = MultipleItemsHelper.remove(values.headOption, cachedData.references)
-
-    customsCacheService
-      .cache[AdditionalFiscalReferencesData](goodsItemCacheId, formId, AdditionalFiscalReferencesData(updatedCache))
-      .map(_ => Redirect(routes.AdditionalFiscalReferencesController.displayPage()))
+    updateCacheModels(itemId, updatedCache, routes.AdditionalFiscalReferencesController.displayPage(itemId))
   }
 
-  private def badRequest(formWithErrors: Form[AdditionalFiscalReference], references: Seq[AdditionalFiscalReference])(
-    implicit request: JourneyRequest[_]
-  ): Result = BadRequest(additional_fiscal_references(formWithErrors, references))
+  private def badRequest(
+    itemId: String,
+    formWithErrors: Form[AdditionalFiscalReference],
+    references: Seq[AdditionalFiscalReference]
+  )(implicit request: JourneyRequest[_]): Result =
+    BadRequest(additional_fiscal_references(itemId, formWithErrors, references))
+
+  private def updateCacheModels(itemId: String, updatedCache: Seq[AdditionalFiscalReference], redirect: Call)(
```

Comment:
Minor but if you called a method called "updateCacheModels" would you expect it to return a `Redirect`?

I'd generally expect it to return an updated model or at the very least `Future[Unit]`

Reply:
commented elsewhere about this.